evap/results/tests/test_exporters.py

```diff
@@ -254,3 +255,39 @@ def test_contributor_result_export(self):
         self.assertEqual(workbook.sheets()[0].row_values(8)[2], 3.0)
         self.assertEqual(workbook.sheets()[0].row_values(10)[0], "Overall Average Grade")
         self.assertEqual(workbook.sheets()[0].row_values(10)[2], 3.25)
+
+    def test_text_answer_export(self):
+        evaluation = baker.make(Evaluation, name_en="eval_en", name_de="eval_de")
+
+        questions = [baker.make(Question, questionnaire__type=t) for t in [10, 20, 30]]
+        for idx in [0, 1, 2, 2, 0]:
+            baker.make(TextAnswer, question=questions[idx], contribution__evaluation=evaluation)
+
+        # The InputData object needs to be hand crafted here, because the
+        # items returned from collect_data(evaluation).contribution_results
+        # did not link back to the questions.
+        # This is probably because baker.make does not fully cover the steps
+        # taken when a user submits something on the website.
```

Comment:
The implementation of `collect_results` iterates like this:
```python
for contribution in evaluation.contributions.all().prefetch_related("questionnaires", "questionnaires__questions"):
    for questionnaire in contribution.questionnaires.all():
        do_stuff™()
```

You use baker to make TextAnswer objects. baker will create a Contribution for each of these. However, you only specified that these newly created contributions should have `evaluation=evaluation`, so their `questionnaires` Many-To-Many-Field will still be empty.

So, in order to fix this, simply change
```python
baker.make(TextAnswer, question=questions[idx], contribution__evaluation=evaluation)
```
to
```python
baker.make(TextAnswer, question=questions[idx], contribution__evaluation=evaluation,
           contribution__questionnaires=[questions[idx].questionnaire])
```

Then, `collect_results(evaluation).contribution_results[0].questionnaire_results` is not empty anymore.